src/Microsoft.ML.CpuMath/CpuMathUtils.netcoreapp.cs

```diff
@@ -9,19 +9,199 @@ namespace Microsoft.ML.Runtime.Internal.CpuMath
+
+            if (iposMin >= iposLim)
+            {
+                if (!add)
+                    dst.ZeroItems();
+                return;
+            }
+
+            Contracts.AssertNonEmpty(rgposSrc);
+
+            if (Sse.IsSupported)
+            {
+                if (!tran)
+                {
+                    Contracts.Assert(crun >= 0);
+                    Contracts.Assert(crun <= dst.Size);
+                    SseIntrinsics.MatMulPA(add, mat, rgposSrc, srcValues, posMin, iposMin, iposLim, dst, crun, srcValues.Size);
+                }
+                else
+                {
+                    Contracts.Assert(crun >= 0);
+                    Contracts.Assert(crun <= srcValues.Size);
+                    SseIntrinsics.MatMulTranPA(add, mat, rgposSrc, srcValues, posMin, iposMin, iposLim, dst, dst.Size);
+                }
+            }
+            else
+            {
+                if (!tran)
+                {
+                    Contracts.Assert(crun >= 0);
+                    Contracts.Assert(crun <= dst.Size);
+                    for (int i = 0; i < crun; i++)
+                    {
+                        float dotProduct = 0;
+                        for (int j = iposMin; j < iposLim; j++)
+                        {
+                            int col = rgposSrc[j] - posMin;
+                            dotProduct += mat[i * srcValues.Size + col] * srcValues[col];
+                        }
+
+                        if (add)
+                        {
+                            dst[i] += dotProduct;
+                        }
+                        else
+                        {
+                            dst[i] = dotProduct;
+                        }
+                    }
+                }
+                else
+                {
+                    Contracts.Assert(crun >= 0);
+                    Contracts.Assert(crun <= srcValues.Size);
+                    for (int i = 0; i < dst.Size; i++)
+                    {
+                        float dotProduct = 0;
+                        for (int j = iposMin; j < iposLim; j++)
+                        {
+                            int col = rgposSrc[j] - posMin;
+                            dotProduct += mat[col * dst.Size + i] * srcValues[col];
+                        }
+
+                        if (add)
+                        {
+                            dst[i] += dotProduct;
+                        }
+                        else
+                        {
+                            dst[i] = dotProduct;
+                        }
+                    }
+
+                }
+            }
+        }
+
+        public static void Add(float a, float[] dst, int count)
+        {
+            Contracts.AssertNonEmpty(dst);
+            Contracts.Assert(count > 0);
+            Contracts.Assert(count <= dst.Length);
+
+            Add(a, new Span<float>(dst, 0, count));
+        }
+
+        // dst += a
```

Comment:
Nit: I think it is pretty clear what this is doing. Maybe this comment can be deleted.